src/Servers/Kestrel/Core/src/Internal/Infrastructure/TransportManager.cs

```diff
@@ -53,6 +53,14 @@ public async Task<EndPoint> BindAsync(EndPoint endPoint, ConnectionDelegate conn
             }
         }
 
+        // Special case situation where a named pipe endpoint is specified and there is no matching transport.
+        // The named pipe transport is only registered on Windows. The test is done at this point so there is
+        // the opportunity for the app to register their own transport to handle named pipe endpoints.
```

Comment:
Should there be a test that registers its own "named pipe" transport to make sure we don't throw?